cli/src/pcluster/api/controllers/cluster_operations_controller.py

```diff
@@ -49,7 +59,7 @@
 def create_cluster(
```

Comment:
In general, why do we re-imlement this logic here rather than reuse (eventually with some changes) the code in the `Cluster` class defined in `cli/src/pcluster/models/cluster.py`? This is the code that is used by the `pcluster_api.py` implementation of `create_cluster`

Reply:
I tried to move to the controller the logic required to handle API specific errors. Also I did not want to heavily modify the cluster.py class at the moment to avoid impacting current CLI functionalities that are not hooked yet to the API call